core/net/rpl/rpl-icmp6.c

```diff
@@ -887,6 +887,7 @@ dao_input_nonstoring(void)
   }
 
   if(flags & RPL_DAO_K_FLAG) {
+    uip_clear_buf();
     dao_ack_output(instance, &dao_sender_addr, sequence);
```

Comment:
There are two more places to call `dao_ack_output()` in rpl-icmp6.c. `dao_input_storing()` calls them.

In order to keep consistency in the source file, it may be nice to put `uip_clear_buf()` before calling each of them. However it works without calling `uip_clear_buf()` there in Storing mode. In Storing mode, a DAO is destined for a link-local address and does not have a Hop-by-Hop option. I did not see any issue on DAO-ACK in testing with Storing mode.